test/unit_tests/CMakeLists.txt

```diff
@@ -28,6 +29,72 @@ set(SRCS
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
+if(WIN32)
+    include(ExternalProject)
+    set(zlib_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/external/zlib_archive)
+    set(ZLIB_INSTALL ${CMAKE_CURRENT_BINARY_DIR}/zlib/install)
+    set(ZLIB_STATIC_LIBRARIES
+            debug ${CMAKE_CURRENT_BINARY_DIR}/zlib/install/lib/zlibstaticd.lib
+            optimized ${CMAKE_CURRENT_BINARY_DIR}/zlib/install/lib/zlibstatic.lib)
+    set(ZLIB_HEADERS
+            "${ZLIB_INSTALL}/include/zconf.h"
+            "${ZLIB_INSTALL}/include/zlib.h")
+    ExternalProject_Add(zlib
+        PREFIX zlib
+        URL https://sourceforge.net/projects/libpng/files/zlib/1.2.11/zlib1211.zip
+        URL_MD5 16b41357b2cd81bca5e1947238e64465
+        INSTALL_DIR ${ZLIB_INSTALL}
+        BUILD_IN_SOURCE 1
+        BUILD_BYPRODUCTS ${ZLIB_STATIC_LIBRARIES}
+        DOWNLOAD_DIR "${DOWNLOAD_LOCATION}"
+        CMAKE_CACHE_ARGS
+            -DCMAKE_BUILD_TYPE:STRING=Release
+            -DCMAKE_INSTALL_PREFIX:STRING=${ZLIB_INSTALL}
+            -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON)
+    add_custom_target(zlib_create_destination_dir
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${zlib_INCLUDE_DIR}
+        DEPENDS zlib)
+    add_custom_target(zlib_copy_headers_to_destination 
+        DEPENDS zlib_create_destination_dir)
+    foreach(header_file ${ZLIB_HEADERS})
+        add_custom_command(TARGET zlib_copy_headers_to_destination PRE_BUILD
+            COMMAND ${CMAKE_COMMAND} -E copy_if_different ${header_file} ${zlib_INCLUDE_DIR})
+    endforeach()
+
+    set(png_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/external/png_archive)
```

Comment:
`Could NOT find PNG (missing: PNG_LIBRARY PNG_PNG_INCLUDE_DIR)`

Why is `PNG_PNG_INCLUDE_DIR`?

Reply:
`find_package` is not even aware of `ExternalProject_Add`. No worries, I'm still working on it.